### PR TITLE
build image from GBI in CI

### PR DESCRIPTION
**Pull request by ikraemer-dd**
I tested this in https://gitlab.ddbuild.io/DataDog/bank-vaults-forked/-/jobs/493240647 successfully.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,58 +1,19 @@
...
-
-WORKDIR /usr/local/src/bank-vaults
-
-ARG GOPROXY
-
+ARG BASE_IMAGE
+# See
+# https://docs.docker.com/engine/reference/builder/#automatic-platform-args-in-the-global-scope
+# for info on BUILDPLATFORM, TARGETOS, TARGETARCH, etc.
+FROM --platform=$BUILDPLATFORM golang:1.22.2 AS builder
```
Comment (maruina) [inline Dockerfile]: I would keep the same go version declared in https://github.com/DataDog/bank-vaults-forked/blob/5fb4a8c0dcc04f38f0b7dfa93966c66112a32e20/go.mod#L3 so in this case `golang:1.21` - any patch version will do. But I see they'te not doing this in the original Dockerfile, so feel free to ignore me :)

**Review by maruina — APPROVED** [review]
LGMT, but can we add a note about why we're changing the Dockerfile so heavily? I assume we don't need a lot of things they're using, but would be great to write down the why as future reference

**Comment by nyodas** [comment]
Another way to think about this might be to keep the original docker file and add another `Dockerfile.dd`.
It would need some change to the gitlab template i assume, to allow for a custom dockerfile.

But it would possibly make upstream PR easier.

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+include: https://gitlab-templates.ddbuild.io/compute-delivery/v2/compute-delivery.yml
+
+test:
+  stage: verify
+  tags: [ "arch:amd64" ]
+  image: registry.ddbuild.io/images/mirror/golang:1.21.5
```
Comment (nyodas) [inline .gitlab-ci.yml]: > [announce.dev] [ACTION REQUIRED] Please upgrade Go to v1.22 by July 31, 2024

Should this be part of the equation here ?

Comment (ikraemer-dd) [reply .gitlab-ci.yml]: Damn, thanks!

**Review by maruina — APPROVED** [review]

**Pull request merged**